src/client.ts

```diff
@@ -1039,6 +1051,10 @@ export class StreamChat<
         tags: ['connection', 'client'],
       });
 
+      this.dispatchEvent({
```

Comment:
Also, if the queryChannels on line 1044 fails, then we don't dispatch connection.recovered event, which our UI sdks rely upon. So we run into situations where network has been recovered, but we never fired connection.recovered event. By letting UI sdk handle state recovery themselves, components can keep track of api failures and display relevant error indicator.